code/modules/mob/living/carbon/human/human.dm

```diff
@@ -659,68 +641,44 @@
 			force_update_limbs()
 			update_body()
 
-/mob/living/carbon/human/proc/set_species(var/new_species, var/default_colour = 1)
-	if(!dna)
-		if(!new_species)
-			new_species = global.using_map.default_species
-	else
-		if(!new_species)
-			new_species = dna.species
-
-	// No more invisible screaming wheelchairs because of set_species() typos.
-	if(!get_species_by_key(new_species))
-		new_species = global.using_map.default_species
-
-	if(dna)
-		dna.species = new_species
+//set_species should not handle the entirety of initing the mob, and should not trigger deep updates
+//It focuses on setting up specie related data, without force applying them uppon organs and the mob's appearence.
+// For transforming an existing mob, look at change_species()
+/mob/living/carbon/human/proc/set_species(var/new_species_name)
+	if(!new_species_name)
+		CRASH("set_species on mob '[src]' was passed a null species name '[new_species_name]'!")
+	var/new_species = get_species_by_key(new_species_name)
+	if(species?.name == new_species_name)
+		return
+	if(!new_species)
+		CRASH("set_species on mob '[src]' was passed a bad species name '[new_species_name]'!")
 
+	//Handle old species transition
 	if(species)
-
-		if(species.name && species.name == new_species)
-			return
-
-		// Clear out their species abilities.
 		species.remove_base_auras(src)
 		species.remove_inherent_verbs(src)
 		holder_type = null
 
-	species = get_species_by_key(new_species)
-	species.handle_pre_spawn(src)
-
-	skin_colour = (species.base_color && default_colour) ? species.base_color : COLOR_BLACK
-
+	//Update our species
+	species = new_species
+	if(dna)
+		dna.species = new_species_name
+	holder_type = null
```

Comment:
This seems to be done twice.

Reply:
You mean the holder_type? Yeah, I'm not entirely sure why.. I'll see what I can do!

Reply:
Removed the extra nulling out of the holder type.